### PR TITLE
feat: Collect lootrun statistics

### PR DESCRIPTION
**Pull request by kristofbolyai**
> But otoh, I think it is okay to add other stats first, so we have more than the measly 2 I created when we start to show anything. - @magicus 

I think this should do it :)

### CODE REVIEW

**Comment by kristofbolyai** [comment]
We don't error for missing i18ns...

**Comment by magicus** [comment]
Should we have a i18n assertion for statistics as well?

**Comment by kristofbolyai** [comment]
> Should we have a i18n assertion for statistics as well?

Most likely yes, I missed them, others will as well.

**Comment by magicus** [comment]
> Most likely yes, I missed them, others will as well.

Fixed in #1863

File: common/src/main/java/com/wynntils/screens/lootrun/WynntilsLootrunsScreen.java
Comment (magicus) [inline common/src/main/java/com/wynntils/screens/lootrun/WynntilsLootrunsScreen.java:1]: This should be WynntilsLootrunPathsScreen, and the package should be lootrunpaths.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/screens/lootrun/WynntilsLootrunsScreen.java:1]: Fixed them

File: common/src/main/java/com/wynntils/models/lootrun/LootrunModel.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    private LootrunFinishedEvent.Completed.Builder lootrunCompletedBuilder;
+    private LootrunFinishedEvent.Failed.Builder lootrunFailedBuilder;
+
+    public LootrunModel() {
+        super(List.of());
+    }
+
+    @SubscribeEvent
+    public void onChatMessage(ChatMessageReceivedEvent event) {
+        StyledText styledText = event.getOriginalStyledText();
```
Comment (magicus) [inline common/src/main/java/com/wynntils/models/lootrun/LootrunModel.java:62]: You should check the recipientType so it is info.

File: common/src/main/java/com/wynntils/models/lootrun/event/LootrunFinishedEvent.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        public int getExperienceGained() {
+            return experienceGained;
+        }
+
+        public int getMobsKilled() {
+            return mobsKilled;
+        }
+
+        public static class Builder {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/models/lootrun/event/LootrunFinishedEvent.java]: The builder idea was nice, but I think you should move them to the model. We have always let the event classes be very lean and not contain anything apart from final fields and getters. It is the responsibility of the model to collect all data needed to be able to construct and send out an event.

**Review by magicus — COMMENTED** [review]
Looks good apart from the builder placement.

File: common/src/main/java/com/wynntils/models/lootrun/event/LootrunFinishedEvent.java (hunk 2 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/models/lootrun/event/LootrunFinishedEvent.java]: I moved it to a new file, I don't like it being in the model.

**Review by magicus — APPROVED** [review]

**Pull request merged**